File: examples/php/database/mysql.php
<?php

declare(strict_types=1);

require_once __DIR__ . '/../vendor/autoload.php';

use Greyfoundry\Identifold\Storage\MySQL\MySQLStorageAdapter;
use Greyfoundry\Identifold\Storage\ReferenceReservation;

$parts = parse_url((string) getenv('IDENTIFOLD_TEST_MYSQL_URL'));
if (!is_array($parts)) {
    throw new RuntimeException('IDENTIFOLD_TEST_MYSQL_URL is required');
}
$connection = new PDO(
    sprintf(
        'mysql:host=%s;port=%d;dbname=%s;charset=utf8mb4',
        $parts['host'],
        $parts['port'],
        ltrim($parts['path'], '/'),
    ),
    $parts['user'],
    $parts['pass'],
    [PDO::ATTR_EMULATE_PREPARES => false, PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION],
);
$adapter = new MySQLStorageAdapter($connection);
$request = new ReferenceReservation(
    '01890f8c-7b2a-7cc3-98b0-112233445568',
    'order',
    'ORD-9876-5432-10-X',
);
$reserved = $adapter->reserve($request);
$mapping = $adapter->resolve($request->reference, $request->namespace);
echo json_encode(['reserved' => $reserved, 'mapping' => $mapping], JSON_THROW_ON_ERROR), PHP_EOL;

File: examples/rust/src/bin/mysql.rs
use identifold::storage::mysql::MySqlStorageAdapter;
use identifold::storage::{ReferenceReservation, StorageAdapter};
use mysql_async::Pool;

#[tokio::main]
async fn main() {
    let database_url =
        std::env::var("IDENTIFOLD_TEST_MYSQL_URL").expect("IDENTIFOLD_TEST_MYSQL_URL is required");
    let pool = Pool::new(database_url.as_str());
    let adapter = MySqlStorageAdapter::new(pool.clone());
    let request = ReferenceReservation {
        machine_id: "01890f8c-7b2a-7cc3-98b0-112233445568".into(),
        namespace: "order".into(),
        reference: "ORD-9876-5432-10-X".into(),
    };
    let reserved = adapter.reserve(&request).await.expect("reserve reference");
    let mapping = adapter
        .resolve(&request.reference, &request.namespace)
        .await
        .expect("resolve reference");
    println!("reserved={reserved} mapping={mapping:?}");
    pool.disconnect().await.expect("disconnect pool");
}
